File: HW/HW11/solution.py
class Solution:
    def flood_fill_help(image, i, j, prev_c, new_c):
        if i < 0 or i >= len(image) or j < 0 or j >= len(image[0]):
            return
        if image[i][j] != prev_c:
            return
        image[i][j] = new_c
        Solution.flood_fill_help(image, i + 1, j, prev_c, new_c)
        Solution.flood_fill_help(image, i - 1, j, prev_c, new_c)
        Solution.flood_fill_help(image, i, j + 1, prev_c, new_c)
        Solution.flood_fill_help(image, i, j - 1, prev_c, new_c)
    
    def flood_fill(self, image, sr, sc, color):
        color_p = image[sr][sc]
        Solution.flood_fill_help(image, sr, sc, color_p, color)
        return image

    
    def printer(array):
        for row in array:
            for col in row:
                print(col, end=" ")
            print()

# Test cases
if __name__ == "__main__":
    test1 = Solution()
    image = [[0, 0, 0], [0, 0, 0]]
    test1.printer(test1.flood_fill(image, 0, 0, 0))

File: HW/HW11/solution.java
package HW.HW11;

public class solution {

    static void floodFillHelp(int image[][], int i, int j, int prevC, int newC) {
        if (i < 0 || i >= image.length || j < 0 || j >= image[0].length)
            return;
        if (image[i][j] != prevC)
            return;
        image[i][j] = newC;
        floodFillHelp(image, i + 1, j, prevC, newC);
        floodFillHelp(image, i - 1, j, prevC, newC);
        floodFillHelp(image, i, j + 1, prevC, newC);
        floodFillHelp(image, i, j - 1, prevC, newC);
    }
    

    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int colorP = image[sr][sc];
        floodFillHelp(image, sr,sc, colorP, color);

        return image;
 
    }

    public static void printer(int[][] array){
        for (int[] i : array){
            for (int j : i){
                System.out.print(j + " ");
            }
            System.out.println();
        }
    }

     //test cases
    public static void main(String[] args) {
    solution test1 = new solution();
    int[][] image = new int[][]{{0,0,0} , {0,0,0}};
    printer(test1.floodFill(image, 0,0,0));
    



    }
     
}
